packages/navitem/src/react/index.tsx

```diff
@@ -0,0 +1,44 @@
+import { PropsWithStylesFor } from '@pluralsight/ps-design-system-util'
+import React from 'react'
+
+import { alignments } from '../vars'
+
+import Context, { ContextValue, initialValue } from './context'
+import { HorzLayout } from './horz'
+import { VertLayout } from './vert'
+
+type AllowedSelectors = 'navitem__bar' | 'navitem__bar--selected'
+
+interface NavItemProps
+  extends PropsWithStylesFor<AllowedSelectors>,
+    Partial<ContextValue> {}
```

Comment:
there is a new util(`PropsWithStylesFor`) to simplify `stylesFor` and add type checking

Reply:
This is a great add, thank you!